BlazarTestBase/src/main/java/com/hubspot/blazar/test/base/data/HikariDataSourceFactory.java

```diff
@@ -0,0 +1,141 @@
+package com.hubspot.blazar.test.base.data;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.sql.Connection;
+import java.time.Duration;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+
+import javax.sql.DataSource;
+
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Throwables;
+import com.mysql.management.MysqldResource;
+import com.mysql.management.MysqldResourceI;
+import com.zaxxer.hikari.HikariConfig;
+
+import io.dropwizard.db.DataSourceFactory;
+import io.dropwizard.db.ManagedDataSource;
+import liquibase.Contexts;
+import liquibase.Liquibase;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.ext.logging.slf4j.Slf4jLogger;
+import liquibase.logging.LogFactory;
+import liquibase.logging.Logger;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import liquibase.resource.ResourceAccessor;
+
+public class HikariDataSourceFactory extends DataSourceFactory {
+  private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(HikariDataSourceFactory.class);
+
+  @Override
+  public ManagedDataSource build(MetricRegistry metricRegistry, String name) {
+
+    final Path baseDir = Paths.get(System.getProperty("java.io.tmpdir"), "mysqld_" + UUID.randomUUID().toString());
+    int port = new Random().nextInt(20000) + 3207;
+
+    final MysqldResource mysqldResource = new MysqldResource(baseDir.toFile());
+
+    Map<String, String> options = new HashMap<>();
+    options.put(MysqldResourceI.PORT, Integer.toString(port));
+    options.put("character-set-server", "utf8");
+
+    /**
+     *  This forces the embedded mysql used for tests to have the same timezone offset
+     *  as the parent java process. This fixes things when you're running
+     *  locally and java thinks its in UTC but mysql uses the system timezone which is not UTC.
+     *
+     *  This also avoids having to install and load TimeZone conversion tables to allow mysql
+     *  to convert `UTC` to `+00:00`. If they are installed into the embedded mysql then you
+     *  can simplify the below to:
+     *
+     *  options.put("default-time-zone", Calendar.getInstance().getTimeZone().getID());
+     */
+
+    Duration offset = Duration.ofMillis(Calendar.getInstance().getTimeZone().getRawOffset());
```

Comment:
My understanding is that we convert / use UTC across the database and in all dates processed in the code (or we should do that and let the clients convert the dates/timestamps to local timezones). Why not just simply set mysql to UTC?